src/libopensc/opensc.h

```diff
@@ -889,6 +896,14 @@ typedef struct sc_context {
 	sc_thread_context_t	*thread_ctx;
 	void *mutex;
 
+#ifdef ENABLE_OPENSSL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+        OSSL_LIB_CTX *osslctx;
+        OSSL_PROVIDER *default_provider;
+        OSSL_PROVIDER *legacy_provider;
```

Comment:
The point of this stuff here is to load these providers in the custom libctx.
providers should not be loaded in the main context and in general loading them randomly in places is not a great thing. We should keep the loading of providers near the libctx creation so that this fact is somewhat under control.

What uses these context is the openssl library, not us directly, we just need to keep handlers to be able to unload them when we free the custom library context

Reply:
The point I was trying to make is that we're calling OpenSSL APIs to *request* the `OSSL_PROVIDER`s without ever using the pointers within OpenSC. But if I get your response right, you're saying that by requesting that pointer, we are actually loading the providers into our custom `OSSL_LIB_CTX`, right?

Reply:
That's correct, as you can see OSSL_PROVIDER_load() takes the custom context as first parameter. So what happens here is that we load those providers in the specific context (instead of the default context when NULL is passed).
And that's the whole point. We keep around the pointers because providers need to be freed later when we destroy the custom context.
See the calls to OSSL_PROVIDER_unload().